### ISSUE TITLE
Alertmanager webhook root status misleading

### ISSUE THREAD

**Issue by josedonizetti** [prometheus/alertmanager#1051]
Alertmanager won't consider the status of the alert when grouping but then sending a webhook notification the root status of the payload is misleading because the group may have both firing and resolved alerts, and the root status only says either one or the other.

Webhook payload
```
{
  "receiver": "webhook",
  "status": "firing",
  "alerts": [
    {
      "status": "firing",
      "labels": {
        "alertname": "ServerStop",
        "env": "prod",
        "status": "firing"
      },
      "annotations": {},
      "startsAt": "2017-10-21T15:50:17.16095407-02:00",
      "endsAt": "0001-01-01T00:00:00Z",
      "generatorURL": ""
    },
    {
      "status": "resolved",
      "labels": {
        "alertname": "ServerStop",
        "env": "prod",
        "instance": "a1",
        "status": "resolved"
      },
      "annotations": {},
      "startsAt": "2017-10-21T15:50:28.806970773-02:00",
      "endsAt": "2017-10-21T15:50:28.806970877-02:00",
      "generatorURL": ""
    }
  ],
  "groupLabels": {
    "alertname": "ServerStop",
    "env": "prod"
  },
  "commonLabels": {
    "alertname": "ServerStop",
    "env": "prod"
  },
  "commonAnnotations": {},
  "externalURL": "http://Joses-MacBook-Pro-2.local:9093",
  "version": "4",
  "groupKey": "{}:{alertname=\"ServerStop\", env=\"prod\"}"
}
```

Alertmanager config:
```
global:
  resolve_timeout: 1h

route:
  group_by: ['alertname', 'env']
  group_wait: 30s
  group_interval: 1m
  repeat_interval: 3h
  receiver: webhook
    
receivers:
- name: webhook
  webhook_configs:
    - url: <url>
```

I'm able two simulate this in master.

**Comment by josedonizetti** [comment]
TBH the way to simulate this 'bug' is very strange, but there are two users complaining about it on the mailing list. But what happens is, two alerts are sent, they match on the group_by clause (same name, and same env), but one of the alerts will be resolved but with a few different labels. By having a few diff labels they won't match on the fingerprint, so one is not resolving the other, they are considered as different alerts.

I was sketching two solutions to this problem:

1) Consider the alert status by default when grouping.
2) Remove the root 'status' of the payload

@stuartnelson3 @brancz What do you think?

**Comment by brian-brazil** [comment]
This is the intended behaviour, status is only resolved if all alerts are resolved.

**Comment by josedonizetti** [comment]
@brian-brazil Indeed. Found the code that does it. Thanks for the help.

For further notice anyone coming to this issue can look into:
https://github.com/prometheus/common/blob/master/model/alert.go#L130

**Comment by ronenn1** [comment]
I've found the reason for this behavior, and a solution.
The root problem was my Prometheus alert rule:
ALERT some_alert
`IF (time() - container_last_seen{name="some_cont",instance="some_fqdn"} > 60) or absent(container_last_seen{name="some_cont",instance="some_fqdn"})
FOR 5m`
Initially, "some_cont" is up so there are no alerts.
Then, I stop the container.
After around a minute (depending on the scrape and evaluate cycles), as expected, Prometheus sets the alert to PENDING, and after few more cycles when the FOR is satisfied, it gets to FIRING and sent to Alertmanager.
So far so good.
Apparently Prometheus keeps the metric for couple of minutes (more then the metrics provider, cAdvisor), and eventually the metric is removed.
Now comes the interesting part.
The first time it evaluate the rule (when the metric doesn't exist) the first condition is probably skipped and the second part is evaluated - and what I believe is happening is this - the metric is absent, but not for 5 minutes, so the alert is changed to PENDING, an alert with status=resolved is sent (in addition to the previous active), and because the metric is missing, there aren't any labels besides the jobs' on this alert. When the absent condition is satisfied again, it gets to FIRING again.
Without the FOR it works fine, but we needed it, so my solution was to change the condition to a single one i.e.
`IF absent(((time() - container_last_seen{name="some_cont",instance="some_fqdn"}) < 60))`
I also needed to explicitly add the `instance` and `name` labels since it is not known to Prometheus when the metric is actually absent (although visible when PENDING).
I don't know if this is a bug.
Hope it helps.
Ronen.
